drivers/SPI.h

```diff
@@ -268,42 +290,34 @@ class SPI : private NonCopyable<SPI> {
     */
     void dequeue_transaction();
     static CircularBuffer<Transaction<SPI>, TRANSACTION_QUEUE_SIZE_SPI> _transaction_buffer;
-#endif
-
-#endif
+#endif // TRANSACTION_QUEUE_SIZE_SPI
 
-public:
-    virtual ~SPI()
-    {
-    }
+#endif // DEVICE_SPI_ASYNCH
 
 protected:
-    spi_t _spi;
+    // holds spi_peripheral_s per peripheral on the device.
+    // Drawback: it costs ram size even if the device is not used.
```

Comment:
This shouldn't be costing space if SPI isn't used at all, I hope. I guess we can live with using more than necessary for buses that aren't used.